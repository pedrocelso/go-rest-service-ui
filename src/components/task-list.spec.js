import React from 'react';
import {shallow} from 'enzyme'

import TaskList from './task-list'

describe(`<TaskList />`, () => {
  it(`Should fetch the tasks from service and render`, () => {
    const service = {
      getTasks: (_) => ({
        fork: (rej, res) => {
          res([
            { id: 10, name: `test`, email: `test@test.com` },
            { id: 11, name: `test1`, email: `test1@test.com` },
          ])
        }
      })
    }

    const component = shallow(
      <TaskList service={service}/>,
    );

    expect(component).toMatchSnapshot();
  });

  it(`Should not display tasks if service returns an error`, () => {
    const service = {
      getTasks: (_) => ({
        fork: (rej, _) => {
          rej(`error fetching tasks`)
        }
      })
    }

    const component = shallow(
      <TaskList service={service}/>,
    );

    expect(component).toMatchSnapshot();
  });
});