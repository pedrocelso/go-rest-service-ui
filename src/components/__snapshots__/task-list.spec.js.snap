// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskList /> Should fetch the tasks from service and render 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="list-group"
  >
    <a
      className="list-group-item list-group-item-action"
      href="#"
      key="10"
    >
      test
       - 
      test@test.com
    </a>
    <a
      className="list-group-item list-group-item-action"
      href="#"
      key="11"
    >
      test1
       - 
      test1@test.com
    </a>
  </div>
</div>
`;

exports[`<TaskList /> Should not display tasks if service returns an error 1`] = `
<div
  className="panel panel-default"
/>
`;
